includes/acl/core/memory.h

```diff
@@ -412,15 +413,37 @@ namespace acl
 	//////////////////////////////////////////////////////////////////////////
 
 	// TODO: get an official L3 cache size
-	constexpr size_t CACHE_FLUSH_BUFFER_BYTES = 12 * 1024 * 1024;
+	constexpr size_t CACHE_FLUSH_BUFFER_BYTES = 20 * 1024 * 1024;
 
-	inline void flush_data_cache(Allocator& allocator)
+	inline uint8_t* allocate_cache_flush_buffer(Allocator& allocator)
 	{
-		uint8_t* buffer = allocate_type_array<uint8_t>(allocator, CACHE_FLUSH_BUFFER_BYTES);
+#if defined(ACL_SSE2_INTRINSICS)
```

Comment:
Good idea but I would go further. Allocate with the alignment of Vector4_32, always and use vector_add or similar when we flush the cache. `data[i] = vector_mul(data[i], vector_set(1.0f));` Eventually we'll overflow and generate infinite/nan but it doesn't matter as long as the assembly isn't stripped out by the compiler.

Reply:
Eventually the math library will be optimized for every platform to use SIMD. The compiler might even be able to auto-vectorize the loop to partially unroll it.